Synthesis failed for nodejs-logging

Hello! Autosynth couldn't regenerate nodejs-logging. :broken_heart:

Here's the output from running `synth.py`:

```
-event <InotifyEvent: src_path=b'./node_modules/.staging/@google-cloud', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@google-cloud>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@grpc', wd=60, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@grpc', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@grpc>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@istanbuljs', wd=61, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@istanbuljs', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@istanbuljs>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@microsoft', wd=62, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@microsoft', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@microsoft>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@opencensus', wd=229, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@opencensus', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@opencensus>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@protobufjs', wd=63, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@protobufjs', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@protobufjs>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@rushstack', wd=248, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@rushstack', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@rushstack>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@sindresorhus', wd=64, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@sindresorhus', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@sindresorhus>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@sinonjs', wd=65, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@sinonjs', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@sinonjs>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@szmarczak', wd=2414, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@szmarczak', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@szmarczak>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@tootallnate', wd=66, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@tootallnate', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@tootallnate>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@types', wd=53, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@types', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@types>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@typescript-eslint', wd=434, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@typescript-eslint', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@typescript-eslint>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@webassemblyjs', wd=2586, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@webassemblyjs', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@webassemblyjs>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@xtuc', wd=2659, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging/@xtuc', wd=52, mask=IN_DELETE|IN_ISDIR, cookie=0, name=@xtuc>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging', wd=52, mask=IN_DELETE_SELF, cookie=0, name=>
DEBUG:watchdog.observers.inotify_buffer:in-event <InotifyEvent: src_path=b'./node_modules/.staging', wd=51, mask=IN_DELETE|IN_ISDIR, cookie=0, name=.staging>
[96msrc/v2/logging_service_v2_client.ts[0m:[93m311[0m:[93m9[0m - [91merror[0m[90m TS2532: [0mObject is possibly 'undefined'.

[7m311[0m         this.descriptors.batching[methodName] ||
[7m   [0m [91m        ~~~~~~~~~~~~~~~~~~~~~~~~~[0m


Found 1 error.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/logging@8.0.3 compile: `tsc -p . && cp -r proto* build/`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/logging@8.0.3 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-08-12T15_33_27_163Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/logging@8.0.3 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/logging@8.0.3 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-08-12T15_33_27_231Z-debug.log
2020-08-12 08:33:27,274 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-logging/synth.py", line 40, in <module>
    node.postprocess_gapic_library()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 193, in postprocess_gapic_library
    install(hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-08-12 08:33:27,446 autosynth [ERROR] > Synthesis failed
HEAD is now at 7c90bbe chore: release 8.0.3 (#868)
Switched to branch 'autosynth'
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 690, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 539, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 670, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 273, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/b742586e-df31-4aac-8092-78288e9ea8e7/targets/github%2Fsynthtool;config=default/tests;query=nodejs-logging;failed=false).

--- a/.kokoro/release/docs-devsite.sh
+++ b/.kokoro/release/docs-devsite.sh
@@ -36,6 +36,11 @@ NAME=$(cat .repo-metadata.json | json name)
 
 mkdir ./_devsite
 cp ./yaml/$NAME/* ./_devsite
+
+# Delete SharePoint item, see https://github.com/microsoft/rushstack/issues/1229
+sed -i -e '1,3d' ./yaml/toc.yml
+sed -i -e 's/^    //' ./yaml/toc.yml
+
 cp ./yaml/toc.yml ./_devsite/toc.yml
 
 # create docs.metadata, based on package.json and .repo-metadata.json.

--- a/.kokoro/release/publish.cfg
+++ b/.kokoro/release/publish.cfg
@@ -61,7 +61,7 @@ build_file: "nodejs-logging/.kokoro/trampoline.sh"
 # Configure the docker image for kokoro-trampoline.
 env_vars: {
     key: "TRAMPOLINE_IMAGE"
-    value: "gcr.io/cloud-devrel-kokoro-resources/node:8-user"
+    value: "gcr.io/cloud-devrel-kokoro-resources/node:12-user"
 }
 
 env_vars: {

--- a/src/v2/config_service_v2_client.ts
+++ b/src/v2/config_service_v2_client.ts
@@ -328,12 +328,11 @@ export class ConfigServiceV2Client {
         }
       );
 
+      const descriptor = this.descriptors.page[methodName] || undefined;
       const apiCall = this._gaxModule.createApiCall(
         callPromise,
         this._defaults[methodName],
-        this.descriptors.page[methodName] ||
-          this.descriptors.stream[methodName] ||
-          this.descriptors.longrunning[methodName]
+        descriptor
       );
 
       this.innerApiCalls[methodName] = apiCall;

--- a/src/v2/logging_service_v2_client.ts
+++ b/src/v2/logging_service_v2_client.ts
@@ -263,7 +263,7 @@ export class LoggingServiceV2Client {
     // requests; denote this.
 
     this.descriptors.batching = {
-      WriteLogEntries: new this._gaxModule.BundleDescriptor(
+      writeLogEntries: new this._gaxModule.BundleDescriptor(
         'entries',
         ['log_name', 'resource', 'labels'],
         null,
@@ -340,12 +340,14 @@ export class LoggingServiceV2Client {
         }
       );
 
+      const descriptor =
+        this.descriptors.page[methodName] ||
+        this.descriptors.batching?.[methodName] ||
+        undefined;
       const apiCall = this._gaxModule.createApiCall(
         callPromise,
         this._defaults[methodName],
-        this.descriptors.page[methodName] ||
-          this.descriptors.stream[methodName] ||
-          this.descriptors.longrunning[methodName]
+        descriptor
       );
 
       this.innerApiCalls[methodName] = apiCall;

--- a/src/v2/metrics_service_v2_client.ts
+++ b/src/v2/metrics_service_v2_client.ts
@@ -305,12 +305,11 @@ export class MetricsServiceV2Client {
         }
       );
 
+      const descriptor = this.descriptors.page[methodName] || undefined;
       const apiCall = this._gaxModule.createApiCall(
         callPromise,
         this._defaults[methodName],
-        this.descriptors.page[methodName] ||
-          this.descriptors.stream[methodName] ||
-          this.descriptors.longrunning[methodName]
+        descriptor
       );
 
       this.innerApiCalls[methodName] = apiCall;

--- a/synth.metadata
+++ b/synth.metadata
@@ -3,23 +3,23 @@
     {
       "git": {
         "name": ".",
-        "remote": "https://github.com/googleapis/nodejs-logging.git",
-        "sha": "3bd950a3bbb3a1bb24165d1b2cb96fc35c06292d"
+        "remote": "git@github.com:googleapis/nodejs-logging",
+        "sha": "7c90bbe93daa32c265e5867f7962aad32407fe6e"
       }
     },
     {
       "git": {
         "name": "googleapis",
         "remote": "https://github.com/googleapis/googleapis.git",
-        "sha": "4f4aa3a03e470f1390758b9d89eb1aa88837a5be",
-        "internalRef": "320300472"
+        "sha": "874846a1917ee5c3fe271449f3cb9a06e75407be",
+        "internalRef": "326288259"
       }
     },
     {
       "git": {
         "name": "synthtool",
         "remote": "https://github.com/googleapis/synthtool.git",
-        "sha": "94421c47802f56a44c320257b2b4c190dc7d6b68"
+        "sha": "5747555f7620113d9a2078a48f4c047a99d31b3e"
       }
     }
   ],
@@ -33,83 +33,5 @@
         "generator": "bazel"
       }
     }
-  ],
-  "generatedFiles": [
-    ".eslintignore",
-    ".eslintrc.json",
-    ".gitattributes",
-    ".github/ISSUE_TEMPLATE/bug_report.md",
-    ".github/ISSUE_TEMPLATE/feature_request.md",
-    ".github/ISSUE_TEMPLATE/support_request.md",
-    ".github/PULL_REQUEST_TEMPLATE.md",
-    ".github/publish.yml",
-    ".github/release-please.yml",
-    ".github/workflows/ci.yaml",
-    ".gitignore",
-    ".jsdoc.js",
-    ".kokoro/.gitattributes",
-    ".kokoro/common.cfg",
-    ".kokoro/continuous/node10/common.cfg",
-    ".kokoro/continuous/node10/docs.cfg",
-    ".kokoro/continuous/node10/lint.cfg",
-    ".kokoro/continuous/node10/samples-test.cfg",
-    ".kokoro/continuous/node10/system-test.cfg",
-    ".kokoro/continuous/node10/test.cfg",
-    ".kokoro/continuous/node12/common.cfg",
-    ".kokoro/continuous/node12/test.cfg",
-    ".kokoro/docs.sh",
-    ".kokoro/lint.sh",
-    ".kokoro/populate-secrets.sh",
-    ".kokoro/presubmit/node10/common.cfg",
-    ".kokoro/presubmit/node10/samples-test.cfg",
-    ".kokoro/presubmit/node10/system-test.cfg",
-    ".kokoro/presubmit/node12/common.cfg",
-    ".kokoro/presubmit/node12/test.cfg",
-    ".kokoro/publish.sh",
-    ".kokoro/release/docs-devsite.cfg",
-    ".kokoro/release/docs-devsite.sh",
-    ".kokoro/release/docs.cfg",
-    ".kokoro/release/docs.sh",
-    ".kokoro/release/publish.cfg",
-    ".kokoro/samples-test.sh",
-    ".kokoro/system-test.sh",
-    ".kokoro/test.bat",
-    ".kokoro/test.sh",
-    ".kokoro/trampoline.sh",
-    ".mocharc.js",
-    ".nycrc",
-    ".prettierignore",
-    ".prettierrc.js",
-    "CODE_OF_CONDUCT.md",
-    "CONTRIBUTING.md",
-    "LICENSE",
-    "README.md",
-    "api-extractor.json",
-    "linkinator.config.json",
-    "protos/google/logging/v2/log_entry.proto",
-    "protos/google/logging/v2/logging.proto",
-    "protos/google/logging/v2/logging_config.proto",
-    "protos/google/logging/v2/logging_metrics.proto",
-    "protos/protos.d.ts",
-    "protos/protos.js",
-    "protos/protos.json",
-    "renovate.json",
-    "samples/README.md",
-    "src/v2/config_service_v2_client.ts",
-    "src/v2/config_service_v2_client_config.json",
-    "src/v2/config_service_v2_proto_list.json",
-    "src/v2/index.ts",
-    "src/v2/logging_service_v2_client.ts",
-    "src/v2/logging_service_v2_client_config.json",
-    "src/v2/logging_service_v2_proto_list.json",
-    "src/v2/metrics_service_v2_client.ts",
-    "src/v2/metrics_service_v2_client_config.json",
-    "src/v2/metrics_service_v2_proto_list.json",
-    "system-test/install.ts",
-    "test/gapic_config_service_v2_v2.ts",
-    "test/gapic_logging_service_v2_v2.ts",
-    "test/gapic_metrics_service_v2_v2.ts",
-    "tsconfig.json",
-    "webpack.config.js"
   ]
 }

